qiskit/aqua/algorithms/phase_estimators/phase_estimator_result.py

```diff
@@ -0,0 +1,158 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Result of running PhaseEstimator"""
+
+from typing import Dict, Union
+import numpy
+from qiskit.result import Result
+
+# Maybe we want to use this abstract class
+from qiskit.aqua.algorithms import AlgorithmResult
+
+
+class PhaseEstimatorResult(AlgorithmResult):
+    """Store and manipulate results from running `PhaseEstimator`.
+
+    This class is instantiated by the `PhaseEstimator` class, not via user code.
+    The `PhaseEstimator` class generates a list of phases and corresponding weights. Upon completion
+    it returns the results as an instance of this class. The main method for accessing the results
+    is `filter_phases`.
+    """
+
+    def __init__(self, num_evaluation_qubits: int,
+                 circuit_result: Result,
+                 phases: Union[numpy.ndarray, Dict[str, float]]) -> None:
+        """
+        Args:
+            num_evaluation_qubits: number of qubits in phase-readout register.
+            circuit_result: result object returned by method running circuit.
+            phases: ndarray or dict of phases and frequencies determined by QPE.
+        """
+        # int: number of qubits in phase-readout register
+        self._num_evaluation_qubits = num_evaluation_qubits
+
+        self._phases = phases
+
+        # result of running the circuit (on hardware or simulator)
+        self._circuit_result = circuit_result
+
+        super().__init__()
+
+    @property
+    def phases(self) -> Union[numpy.ndarray, dict]:
+        """Return all phases and their frequencies computed by QPE.
+
+        This is an array or dict whose values correspond to weights on bit strings.
+        """
+        return self._phases
+
+    @property
+    def circuit_result(self) -> Result:
+        """Return the result object returned by running the QPE circuit (on hardware or simulator).
+
+        This is useful for inspecting and troubleshooting the QPE algorithm.
+        """
+        return self._circuit_result
+
+    def single_phase(self) -> float:
```

Comment:
I think this should also be a property, otherwise the UX is different than retrieving other results. Also should this be `single_phase` or `most_likely_phase`?